commands/command_lock.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/github/git-lfs/api"
+	"github.com/github/git-lfs/config"
+	"github.com/github/git-lfs/git"
+	"github.com/spf13/cobra"
+)
+
+var (
+	lockRemote     string
+	lockRemoteHelp = "specify which remote to use when interacting with locks"
+
+	// TODO(taylor): consider making this (and the above flag) a property of
+	// some parent-command, or another similarly less ugly way of handling
+	// this
+	setLockRemoteFor = func(c *config.Configuration) {
+		c.CurrentRemote = lockRemote
+	}
+
+	lockCmd = &cobra.Command{
+		Use: "lock",
+		Run: lockCommand,
+	}
+)
+
+func lockCommand(cmd *cobra.Command, args []string) {
+	setLockRemoteFor(config.Config)
+
+	if len(args) == 0 {
+		Print("Usage: git lfs lock <path>")
+		return
+	}
+
+	latest, err := git.CurrentRemoteRef()
+	if err != nil {
+		Error(err.Error())
+		Exit("Unable to determine lastest remote ref for branch.")
+	}
+
+	path, err := lockPath(args[0])
+	if err != nil {
+		Error(err.Error())
+	}
+
+	s, resp := API.Locks.Lock(&api.LockRequest{
+		Path:               path,
+		Committer:          api.CurrentCommitter(),
+		LatestRemoteCommit: latest.Sha,
```

Comment:
I'm still dubious about why you use the latest *remote* commit here. This is not necessarily the state of the local file. If the server agrees that this is the latest commit known and grants the lock, but this remote ref is *ahead* of the current HEAD and the file is changed, locking it in its current state and allowing the edit is guaranteed to cause a merge conflict, which is what we're trying to avoid. 

Really you need to do one of the following:

1. Pass the current HEAD commit SHA. Server should reject the lock unless it's the latest one it knows about which includes a change to this file. Or,
1. Pass the latest known remote ref as here, but the *client* has to check whether there are changes to this file between HEAD and that remote ref. At this point the user either needs to update their HEAD as a whole, or cherry pick the file from that remote ref into HEAD, or accept that they will create a conflict.